src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function validateLengthes(...lengthes: number[]): void {
+  const maxSide = Math.max(...lengthes);
+  const sidesSum = lengthes.reduce((a, b) => a + b, 0);
+  const hasNullSize = lengthes.some((number) => number <= 0);
+  const hasImpossibleSize = maxSide >= sidesSum - maxSide;
+
+  if (hasNullSize || (lengthes.length === 3 && hasImpossibleSize)) {
+    throw new Error('impossible sides of figure');
+  }
 }
```

Comment:
Looks very complicated. If you do it in constructor this will look simpler and on its place. Plus checklist said that we must describe error when we throw one. For example `throw new Error('each side must be greater than zero');` or `throw new Error('radius must be greater than zero');`